testsuites/syncgateway/functional/tests/test_log_rotation_new.py

```diff
@@ -0,0 +1,609 @@
+
+    sg_helper.stop_sync_gateways(cluster_config=cluster_conf, url=sg_one_url)
+    # Change the timestamps for SG logs when SG stopped (Name is unchanged)
+    for log in SG_LOGS_MAXAGE:
+        file_name = "/tmp/sg_logs/{}.log".format(log)
+        remote_executor.execute("sudo touch -d \"{} days ago\" {}".format(SG_LOGS_MAXAGE[log], file_name))
+
+    sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=temp_conf)
+
+    for log in SG_LOGS_MAXAGE:
+        _, stdout, _ = remote_executor.execute("ls /tmp/sg_logs/ | grep {} | wc -l".format(log))
+        # Verify that new log file was not created
+        assert stdout[0].rstrip() == '2'
+
+    # Remove generated conf file
+    os.remove(temp_conf)
+
+
+# https://github.com/couchbase/sync_gateway/issues/2221
+@pytest.mark.syncgateway
+@pytest.mark.logging
+@pytest.mark.parametrize("sg_conf_name", ["log_rotation_new"])
+def test_log_rotation_invalid_path(params_from_base_test_setup, sg_conf_name):
+    """Test to check that SG is not started with invalid logFilePath.
+    OS specific case. SG should check if path correct on startup
+    """
+    cluster_conf = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    log_info("Using cluster_conf: {}".format(cluster_conf))
+    log_info("Using sg_conf: {}".format(sg_conf))
+
+    cluster_helper = ClusterKeywords()
+    cluster_hosts = cluster_helper.get_cluster_topology(cluster_conf)
+    sg_admin_url = cluster_hosts["sync_gateways"][0]["admin"]
+    sg_ip = host_for_url(sg_admin_url)
+
+    if get_sync_gateway_version(sg_ip)[0] < "2.1":
+        pytest.skip("Test NA for SG  < 2.1")
+
+    cluster = Cluster(config=cluster_conf)
+    cluster.reset(sg_config_path=sg_conf)
+
+    sg_one_url = cluster_hosts["sync_gateways"][0]["public"]
+
+    # read sample sg_conf
+    data = load_sync_gateway_config(sg_conf, cluster_hosts["couchbase_servers"][0], cluster_conf)
+
+    # set non existing logFilePath
+    data['logging']["log_file_path"] = "/12345/1231/131231.log"
+    # create temp config file in the same folder as sg_conf
+    temp_conf = "/".join(sg_conf.split('/')[:-2]) + '/temp_conf.json'
+
+    with open(temp_conf, 'w') as fp:
+        json.dump(data, fp, indent=4)
+
+    # Stop sync_gateways
+    log_info(">>> Stopping sync_gateway")
+    sg_helper = SyncGateway()
+
+    sg_helper.stop_sync_gateways(cluster_config=cluster_conf, url=sg_one_url)
+    try:
+        sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=temp_conf)
+    except ProvisioningError:
+        sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=sg_conf)
+        # Remove generated conf file
+        os.remove(temp_conf)
+        return
+
+    # Remove generated conf file
+    os.remove(temp_conf)
+    pytest.fail("SG shouldn't be started!!!!")
```

Comment:
this should be inside except block. if start sync_gateway throws error, it should fail. Adding here will always fail the test, right?

Reply:
No, the pass criteria is that SG fails to start, in which case, it will go into the except. If it doesn't, it means SG started and we mark it fail